testsuite/cluster-test/src/effects/mod.rs

```diff
@@ -9,50 +9,21 @@ mod reboot;
 mod remove_network_effects;
 mod stop_container;
 
-use crate::thread_pool_executor::ThreadPoolExecutor;
 use failure;
+use futures::future::BoxFuture;
 pub use network_delay::three_region_simulation_effects;
 pub use network_delay::NetworkDelay;
 pub use packet_loss::PacketLoss;
 pub use reboot::Reboot;
 pub use remove_network_effects::RemoveNetworkEffects;
-use slog_scope::info;
 use std::fmt::Display;
 pub use stop_container::StopContainer;
+
 pub trait Action: Display + Send {
-    fn apply(&self) -> failure::Result<()>;
-    fn is_complete(&self) -> bool;
+    fn apply(&self) -> BoxFuture<failure::Result<()>>;
```

Comment:
Why not just `fn apply(&self) -> Future<failure::Result<()>>;`

Reply:
That won't compile :)